universal-application-tool-0.0.1/app/views/components/FieldWithLabel.java

```diff
@@ -195,6 +195,8 @@ public ContainerTag getContainer() {
       if (this.fieldValueNumber.isPresent()) {
         fieldTag.withValue(String.valueOf(this.fieldValueNumber.getAsLong()));
       }
+      // We only allow integer input.
+      fieldTag.attr("oninput", "this.value=(parseInt(this.value)||0)");
```

Comment:
question: Does this mean that if parseInt doesn't work then we fill it with 0 instead? I'm slightly concerned with this approach if so, but I might be reading it wrong. :)

Reply:
You are right. parseInt parses until an illegal character (e.g. non-digit) and truncates the rest of the string: so `12a34` is parsed as `12`. Here, the result is that the user will not be able to input `a` character in the first place, which seems a reasonable experience to me.

The only case where parseInt returns a `NaN` (it doesn't fail) is when there is an empty string. I think it is reasonable to set the value to 0 in this situation. The result is that once the user tries to input something, they can never set it back to empty string, but that also feels like a reasonable experience to me.

Reply:
Ahh, the first paragraph sounds great to me. Just wanted to make sure that e.g. "50.0" would return 50 rather than 0.

As to empty string --> 0... sounds a little less ideal, especially that you can never "clear out" an answer with this. Can we special case empty string? We need the ability for number fields to be able to be optional anyway.

Reply:
What do you think? If it makes it really difficult to deal with then don't worry about it. But I think allowing empty inputs is probably desirable.

Reply:
Yes, "50.0" would be 50.

Yeah, I also don't like losing the ability to have empty string. I think I have something working.